app/controllers/application_controller.rb

```diff
@@ -1,2 +1,19 @@
 class ApplicationController < ActionController::Base
+  before_action :authorize
+
+  private def redirect_to_homepage_if_logged_in
+    return unless session[:user_id]
+
+    redirect_to store_index_path, notice: t('notice.application.logged_in')
+  end
+
+  private def authorize
+    return if current_user
+
+    redirect_to login_url, alert: t('notice.application.login_request')
```

Comment:
you have used login_url and login_path.
Is there any difference?
If not, use same method everywhere.